### PR TITLE
Append Source Address TLV to Child Update Request if short address is known.

### PR DESCRIPTION
**Pull request by hubertmis**
This PR should fix certification tests 6.5 in the new harness (beta 3).

### CODE REVIEW

File: src/core/thread/mle.cpp [hunk 1]
```diff
@@ -1560,6 +1560,7 @@ ThreadError Mle::SendChildUpdateRequest(void)
     VerifyOrExit((message = NewMleMessage()) != NULL, ;);
     SuccessOrExit(error = AppendHeader(*message, Header::kCommandChildUpdateRequest));
     SuccessOrExit(error = AppendMode(*message, mDeviceMode));
+    SuccessOrExit(error = AppendSourceAddress(*message));
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: This change adds the Source Address TLV when the Child is in the Detached state.  Maybe I missed it, but I could not find where the Thread spec states this requirement.

Comment (hubertmis) [inline src/core/thread/mle.cpp]: Thread spec is not clear in this case. I've found it in 4.5.1 and 4.12.3 but I'm now sure if it applies to this case.
Certification harness checks if Source Address TLV is present in Child Update sent by a detached child. That's why I added it but I'm not sure if harness is compliant with the spec.

Comment (bukepo) [inline src/core/thread/mle.cpp]: We also encountered this issue and reported it to Certification Harness. They've fixed it in newer version.

Comment (jwhui) [inline src/core/thread/mle.cpp]: I believe this check was erroneously added in V39 and is tracked in [DEV-1358](https://threadgroup.atlassian.net/browse/DEV-1358?jql=project%20%3D%20DEV%20AND%20text%20~%20%22Source%20Address%20TLV%22).  @librasungirl, can you confirm?

Comment (hubertmis) [inline src/core/thread/mle.cpp]: In this case I'll close this PR.

Comment (librasungirl) [inline src/core/thread/mle.cpp]: Yes, it is a known issue in internal V39 which has been fixed in internal V40 Test Harness.

**Pull request closed without merging**